### PR TITLE
Suite to test compression and encryption with S/W QAT enabled on the …

### PR DESCRIPTION
**Pull request by TejasC88**
…cluster

Tests added :
- Compression with all supported algorithms
- Per object encryption and per bucket encryption with KMS Vault

Pass log : http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/cephci-run-1FZWIN/

### CODE REVIEW

**Review by anrao19 — APPROVED** [review]
LGTM

File: suites/reef/rgw/tier-1_rgw_sw_qat.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# This test suite tests the QAT S/w feature provided by the intree QAT drier on RHEL 9.4
```
Comment (viduship) [inline suites/reef/rgw/tier-1_rgw_sw_qat.yaml:1]: spellcheck, please fix the words

File: suites/reef/rgw/tier-1_rgw_sw_qat.yaml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      desc: test compresstion with snappy type
+      polarion-id: CEPH-11350
+      module: sanity_rgw.py
+      config:
+        script-name: test_Mbuckets_with_Nobjects.py
+        config-file-name: test_Mbuckets_with_Nobjects_compression_snappy.yaml
+        timeout: 300
+
+  - test:
+      name: compresstion_with_snappy_type
```
Comment (viduship) [inline suites/reef/rgw/tier-1_rgw_sw_qat.yaml:124]: please fix the name here snappy to zlib

File: suites/reef/rgw/tier-1_rgw_sw_qat.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        timeout: 300
+
+  - test:
+      config:
+        script-name: test_sse_s3_kms_with_vault.py
+        config-file-name: test_sse_s3_per_bucket_encryption_normal_object_upload.yaml
+        timeout: 300
+      desc: test_sse_s3_per_bucket_encryption_normal_object_upload
+      module: sanity_rgw.py
+      name: sse-s3 per bucket encryption test on a regular bucket
```
Comment (viduship) [inline suites/reef/rgw/tier-1_rgw_sw_qat.yaml:140]: Please add a test case with multipart coverage 
https://github.com/red-hat-storage/ceph-qe-scripts/blob/master/rgw/v2/tests/s3_swift/configs/test_sse_s3_per_bucket_encryption_multipart_object_upload.yaml

File: suites/reef/rgw/tier-1_rgw_sw_qat.yaml (hunk 1 above)
Comment (TejasC88) [reply suites/reef/rgw/tier-1_rgw_sw_qat.yaml]: Done :)

File: suites/reef/rgw/tier-1_rgw_sw_qat.yaml (hunk 2 above)
Comment (TejasC88) [reply suites/reef/rgw/tier-1_rgw_sw_qat.yaml]: Done .

File: suites/reef/rgw/tier-1_rgw_sw_qat.yaml (hunk 3 above)
Comment (TejasC88) [reply suites/reef/rgw/tier-1_rgw_sw_qat.yaml:140]: @viduship Ive added a per bucket encryption test for multipart , pls review .

**Review by viduship — APPROVED** [review]

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**